tasks/process-spectrum.js

```diff
@@ -146,698 +129,573 @@ function conditionSelector(newSelector) {
-                        if (
-                            component.hoist ||
-                            (typeof component.hoist === 'undefined' &&
-                                component.replace.type === 'attribute') ||
-                            replacenentIsHost
-                        ) {
-                            if (!host) {
-                                host = {
-                                    type: 'pseudo-class',
-                                    kind: 'host',
-                                };
-                            }
-                            if (replacenentIsHost) {
-                                if (component.replace.selectors?.length) {
-                                    host.selectors = host.selectors || [];
-                                    host.selectors.push(
-                                        ...component.replace.selectors
-                                    );
-                                }
-                            } else {
-                                host.selectors = host.selectors || [];
-                                host.selectors.push({
-                                    ...component.replace,
-                                });
-                                if (component.emphasize) {
-                                    host.selectors.push({
-                                        ...component.replace,
-                                    });
-                                }
-                            }
-                        } else {
-                            if (component.replace.type !== 'combinator') {
-                                newSelector.push({
-                                    ...component.replace,
-                                });
-                            } else {
-                                // Assume that the last combinator in is the "correct" combinator.
-                                // This is due to "left over" combinators when hoisting components.
-                                const hasCombinatorLast =
-                                    newSelector.at(-1)?.type === 'combinator';
-                                if (hasCombinatorLast) {
-                                    newSelector.pop();
-                                }
-                                newSelector.push({
-                                    ...component.replace,
-                                });
-                            }
-                        }
-                    }
-                });
+        }
+        matches[index] = match;
+    });
 
-                if (host) {
-                    if (
-                        newSelector.length &&
-                        isHoistedPseudoClass(newSelector[0])
-                    ) {
+    return matches;
+}
+
+/**
+ * Process the selectors array and return the processed selectors
+ * @param {import('./spectrum-css-converter').ProcessedSelector[]} metadata
```

Comment:
there is no exported type `ProcessedSelector` in this repo

Reply:
See my note above - I've updated these to "any" for now since their previous state was undocumented. We can type them more accurately later.